### ISSUE TITLE
Very small prediction probabilities when using Multilabel Text Classification with Classweights

### ISSUE THREAD

**Issue by lady-vanessa** [deepset-ai/FARM#443]
**Question**
Hi there,
i have trained two models, which differ in that one uses class_weights and the other does not.
Without class_weights I get "normal" probabilities per class in the prediction. 
If I use class_weights, my probabilities per class are so small that not a single class scores (the probabilities are at 0.00something).
I wanted to ask, what could be the reason for this and if this is algorithmically intended.
I have to say that my data is relatively unbalanced. So I thought that adding class_weights would improve my model.
Any help is appreciated :)

**Comment by Timoeller** [comment]
Hey @lady-vanessa , do you know the class distribution or class weights that are used? FARM should print out an info about the class weights used.

In general I would say if the class weights are very extreme it is difficult for the model to learn, especially in a multilabel setting.

**Comment by lady-vanessa** [comment]
Hi @Timoeller ,
thanks for your quick response!
I have around 100 classes and the class_weights are in a relatively large range (between min. 0.03 and max. 215.28). 
Is it possible to estimate whether a long training can lead to any success here? Or is it more advisable to abandon the class weights?

**Comment by Timoeller** [comment]
TL;DR I would not use class weights in such an unbalanced scenario. Learning is likely not stable with these class weights.


-----------------

To give a full answer I would need to better understand your use case.
I have some difficulty understanding the necessity of class weights in a multi-label setting anyway. At prediction time you will get a value for each class that is between 0 and 1. So for classifying if that label is present you set thresholds. These thresholds could be chosen per class according to your use case (e.g. lower threshold for rare but important classes). Does that make sense?

**Comment by lady-vanessa** [comment]
Hi @Timoeller ,
thank you very much for your answer. 
I think I was on the wrong track and your explanation helped me a lot! 
I go without class weights and try to adjust the thresholds accordingly :)
Thank you!
